tests/GuardTest.php

```diff
@@ -279,43 +279,35 @@ public function testEnforceStorageLimitWithIterator()
 
     public function testTokenIsRemovedFromStorageWhenPersistentModeIsOff()
     {
-        $self = $this;
-
         $storage = [
             'test_name' => 'test_value123',
         ];
-        $responseFactoryProphecy = $this->prophesize(ResponseFactoryInterface::class);
-        $handler = function () use ($self, &$called) {
-            $responseProphecy = $self->prophesize(ResponseInterface::class);
-            return $responseProphecy->reveal();
-        };
-        $mw = new Guard($responseFactoryProphecy->reveal(), 'test', $storage, $handler);
 
-        $requestProphecy = $this->prophesize(ServerRequestInterface::class);
-        $requestProphecy
-            ->getMethod()
-            ->willReturn('POST')
-            ->shouldBeCalledOnce();
+        $response = $this->createMock(ResponseInterface::class);
+        $requestHandler = $this->createMock(RequestHandlerInterface::class);
+        $requestHandler->method('handle')->willReturn($response);
 
-        $requestProphecy
-            ->withAttribute(Argument::type('string'), Argument::type('string'))
-            ->willReturn($requestProphecy->reveal())
-            ->shouldBeCalledTimes(2);
+        $responseFactory = $this->createMock(ResponseFactoryInterface::class);
 
-        $requestProphecy
-            ->getParsedBody()
-            ->willReturn([
-                'test_name' => 'test_name123',
-                'test_value' => 'invalid_value',
-            ])
-            ->shouldBeCalledOnce();
+        $handler = function () use ($response) {
+            return $response;
+        };
 
-        $requestHandlerProphecy = $this->prophesize(RequestHandlerInterface::class);
+        $mw = new Guard($responseFactory, 'test', $storage, $handler);
 
-        $mw->process($requestProphecy->reveal(), $requestHandlerProphecy->reveal());
-        $this->assertArrayNotHasKey('test_name123', $storage);
+        $request = $this->createMock(ServerRequestInterface::class);
+        $request->expects(self::once())->method('getMethod')->willReturn('POST');
+        $request->expects(self::exactly(2))->method('withAttribute')->willReturn($request);
+        $request->expects(self::once())->method('getParsedBody')->willReturn([
+           'test_name' => 'test_name',
+           'test_value' => 'test_value123',
+        ]);
```

Comment:
Can we use prophecies instead of mocks for cohesion as all the other tests are written that way

Reply:
Sure. I didn't as PHPUnit deprecated Prophecy in [issue 4141](https://github.com/sebastianbergmann/phpunit/issues/4141), so it seems possible that we'll look to move to stock PHPUnit mocks over time.

Maybe that's a separate PR to re-write the lot though.

Reply:
I don't think rewriting all the tests would be time well spent, there's a shim that addresses this deprecation which we've implemented on all the other repos. We will need to add it here as well!

Reply:
Unit test now uses Prophecy